### PR TITLE
Add integration test framework and documentation into RKE2

### PR DESCRIPTION
**Pull request by dereknola**
#### Proposed Changes ####
- Example integration test into RKE2 (etcd snapshot test)

#### Types of Changes ####
New Feature

#### Verification ####
See `testing.md` in `developer docs` to run tests.
Use `make dev-shell` and launch and rke2 server
Once server has started, launch tests via `go tests ./tests/`

#### Linked Issues ####
https://github.com/rancher/rke2/issues/1611

#### Further Comments ####
As RKE2 does not launch as fast as K3s, it was decided that having each test launch and kill a RKE2 server would be time prohibited. It is left to the user to deploy the appropriate server for CLI testing.

### CODE REVIEW

File: tests/util/cmd.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+func Rke2ServerArgs() []string {
+	results, err := Rke2Cmd("kubectl", "get", "nodes", "-o", `jsonpath='{.items[0].metadata.annotations.rke2\.io/node-args}'`)
+	if err != nil {
+		return nil
+	}
+	res := strings.ReplaceAll(results, "'", "")
+	var args []string
+	if err := json.Unmarshal([]byte(res), &args); err != nil {
+		logrus.Error(err)
+		return nil
```
Comment (briandowns) [inline tests/util/cmd.go:94]: Should we return the error here rather than return an empty slice?

File: tests/util/cmd.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+		"kube-scheduler-rke2-server",
+		"rke2-coredns-rke2-coredns",
+	}
+	pods, err := Rke2Cmd("kubectl", "get", "pods", "-A")
+	if err != nil {
+		return false
+	}
+	for _, pod := range podsToCheck {
+		reg := pod + ".+Running"
+		match, err := regexp.MatchString(reg, pods)
```
Comment (briandowns) [inline tests/util/cmd.go:56]: What happens to this err value?

Comment (dereknola) [reply tests/util/cmd.go:56]: Err now gets logged

File: tests/util/cmd.go (hunk 1 above)
Comment (dereknola) [reply tests/util/cmd.go:94]: function now returns err and caller deals with result properly.

**Review by brandond — APPROVED** [review]

**Review by briandowns — APPROVED** [review]
LGTM

**Pull request merged**